src/SmartComponents/Inventory/EntityTable.js

```diff
@@ -87,20 +112,46 @@ class EntityTable extends React.Component {
         };
     }
 
+    buildCells = (item) => {
+        const { columns, showHealth } = this.props;
+        let actions; let health; let cells;
+        if (!item.hasOwnProperty('isOpen')) {
+            actions = this.actionsColumn(item);
+            health = showHealth && this.healthColumn(item);
+            cells = columns.map(({ key, composed, isTime }) => this.renderCol(item, key, composed, isTime));
+        } else {
+            cells = [{
+                title: item.title,
+                colSpan: columns.length + 1 + showHealth
+            }];
+        }
+
+        return [
+            ...cells,
+            health,
+            actions
+        ].filter(Boolean);
+    }
+
     render() {
-        const { columns, entities, rows, showHealth, loaded } = this.props;
+        const { columns, entities, rows, showHealth, loaded, expandable, onExpandClick } = this.props;
         const filteredData = (entities || rows).filter(oneRow => oneRow.account);
```

Comment:
I believe in this case, the OR statement does no longer apply. It's either entities (if we're using a regular, non tree-like table), or rows (if we're using a expandable table). `entities` contains the data from the inventory, so it won't contain any additional collapsible rows. This means that even if we provide the `rows` with extra info, they will get discarded in favor of entities, rendering the 'expandable' attribute useless. See:

![screenshot from 2018-12-28 11-00-36](https://user-images.githubusercontent.com/598891/50512070-61916e00-0a90-11e9-8f5d-783a8a0721d8.png)

Reply:
Hmm.. on the other hand, entities contains valuable info, such as 'facts'. Maybe the OR could be reversed and as a user I could merge 'entities facts' in 'rows' on the reducer.

Reply:
Those entities and rows are something which needs to be refactored, because both holds same information - data from server. Entites were ment to be filtered, sorted paginated data from server. But since we are doing pagination, filter and sort on server that is no longer applicable and items should be removed from store.

Reply:
Do you plan to do that refactor here? I am wondering how to show expandable rows otherwise - so far I locally made the change `(rows || entities)` and I'm using my locally modified library.